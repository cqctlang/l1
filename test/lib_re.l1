

@include <re.cqct>




printf("should be 1: %d\n",re_match("a","a"));
printf("should be 1: %d\n",re_match("[ab]","a"));
printf("should be 1: %d\n",re_match("[ab]*","ababababaaaa"));
printf("should be 1: %d\n",re_match("[ab][bc]*","bbbbcccc"));
printf("should be 0: %d\n",re_match("[ab][bc]*","cbbbcccc"));
printf("should be 1: %d\n",re_match("[ab]*c*","cccc"));
printf("should be 1: %d\n",re_match("[ab]**[bc]*","ccccbbbb"));
printf("should be 0: %d\n",re_match("(ab)c","ac"));
printf("should be 1: %d\n",re_match("(ab)c","abc"));
printf("should be 1: %d\n",re_match("(ab)*c","ababababc"));
printf("should be 1: %d\n",re_match("(ab)*c","c"));
printf("should be 0: %d\n",re_match("(ab)*c","ac"));
printf("should be 0: %d\n",re_match("\\[ab\\]","a"));
printf("should be 1: %d\n",re_match("\\[[ab]\\]","[a]"));
printf("should be 1: %d\n",re_match("\\[ab\\]*","[ab]]]]]]"));
printf("should be 1: %d\n",re_match("abcd","abcd"));
printf("should be 0: %d\n",re_match("abcd","axcd"));
printf("should be 1: %d\n",re_match(".bcd","abcd"));
printf("should be 1: %d\n",re_match("a*bcd","aaaabcd"));
printf("should be 1: %d\n",re_match(".*b.d","aaaabcd"));
printf("should be 1: %d\n",re_match("....b\\*d","aaaab*d"));
printf("should be 0: %d\n",re_match("....b\\*d","aaaabxd"));
printf("should be 0: %d\n",re_match("..y.b.*d","aaaabxd"));
printf("should be 0: %d\n",re_match("...\\.b.d","aaaabxd"));
printf("should be 1: %d\n",re_match("...\\.b.d","aaa.bxd"));
if(0)applyk(@lambda() {
	re_match("...\\.b.d\\","aaaa.bxd");
},@lambda(r) {
	printf("FAIL shouldn't see this!\n");
},@lambda() {
	printf("successfully caused an error\n");
},[]);
if(0)applyk(@lambda() {
	re_match("*...\\.b.d","aaa.bxd");
},@lambda(r) {
	printf("FAIL shouldn't see this!\n");
},@lambda() {
	printf("successfully caused an error\n");
},[]);

printf("should be 1: %d\n",re_match("a?","a"));
printf("should be 1: %d\n",re_match("aa?","a"));
printf("should be 1: %d\n",re_match("ab?","a"));
printf("should be 1: %d\n",re_match("a?b?","a"));
printf("should be 1: %d\n",re_match("a?b?","ab"));
printf("should be 1: %d\n",re_match("a?b?",""));
printf("should be 0: %d\n",re_match("a?b?c",""));
printf("should be 0: %d\n",re_match("a?b?c","aaaabc"));
printf("should be 0: %d\n",re_match("a?a?a?a?a*c","aaaaaaaaaaaaaaabc"));
printf("should be 1: %d\n",
		re_match("a?a?a?a?a?a?a?a?a?a?a?a?a?a?a?a?aaaaaaaaac",
			 "aaaaaaaaaaaaaaac"));
printf("should be 1: %d\n",
		re_match("a?a?a?a?a?a?a?a?a?a?a?a?a?a?a?a?c*",
			 ""));
printf("should be 1: %d\n",re_match("\000*a","\000\000\000a"));

printf("should be 5: %d\n",re_find("aaa","ababbaaaxxxxxaaa"));
printf("should be -1: %d\n",re_find("aaab","ababbaaaxxxxxaaa"));
printf("should be 5: %d\n",re_find("aaa?x","ababbaaaxxxxxaaa"));
printf("should be 6: %d\n",re_find("aa?x*aaa","ababbaaaxxxxxaaa"));
printf("should be 12: %d\n",re_find("[xy]a","ababbaaaxxxxxaaa"));
printf("should be 0: %d\n",re_find(".","ababbaaaxxxxxaaa"));
printf("should be -1: %d\n",re_find(".......................","ababbaaaxxxxxaaa"));
printf("should be 0: %d\n",re_find("(ababba)|(babbaa)","ababbaaaxxxxxaaa"));
printf("should be 5: %d\n",re_find("(xbabba)|(aa*x)x","ababbaaaxxxxxaaa"));

printf("should be 0: %d\n",re_findend("a","aaaaa"));
printf("should be 1: %d\n",re_findend("aa","aaaaa"));
printf("should be 4: %d\n",re_findend("aa*","aaaaa"));
printf("should be -1: %d\n",re_findend("aa*b","aaaaa"));
printf("should be 4: %d\n",re_findend("ba*","abaaa",1));
printf("should be -1: %d\n",re_findend("ba*","abaaa",2));

