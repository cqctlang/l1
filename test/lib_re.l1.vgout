entry: var params, 8 locals, 0 temps
tokenize: 1 params, 18 locals, 0 temps
re_create: 1 params, 3 locals, 0 temps
L333: 2 params, 17 locals, 0 temps
node_str: var params, 9 locals, 0 temps
nxt_nodes: 2 params, 13 locals, 0 temps
dfa_match: 2 params, 7 locals, 0 temps
dfa_match_helper: 3 params, 14 locals, 0 temps
L2322: 2 params, 6 locals, 0 temps
L2348: 1 params, 6 locals, 0 temps
L2445: 2 params, 6 locals, 0 temps
L2470: 1 params, 6 locals, 0 temps
re_match: 2 params, 1 locals, 0 temps
re_find: 2 params, 11 locals, 0 temps
L2680: 0 params, 0 locals, 0 temps
L2684: 1 params, 0 locals, 0 temps
L2688: 0 params, 0 locals, 0 temps
L2698: 0 params, 0 locals, 0 temps
L2702: 1 params, 0 locals, 0 temps
L2706: 0 params, 0 locals, 0 temps
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 5: 5
should be -1: -1
should be 5: 5
should be 6: 6
