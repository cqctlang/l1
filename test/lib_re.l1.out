should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 5: 5
should be -1: -1
should be 5: 5
should be 6: 6
should be 12: 12
should be 0: 0
should be -1: -1
should be 0: 0
should be 5: 5
should be 0: 0
should be 1: 1
should be 4: 4
should be -1: -1
should be 4: 4
should be -1: -1
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 1: 1
should be 0: 0
should be 1: 1
should be 1: 1
should be 1: 1
should be 1: 1
should be 0: 0
should be 0: 0
should be 0: 0
